### PR TITLE
Detect method varargs changes

### PR DESCRIPTION
**Pull request by Marcono1234**
Changing a method from varargs to non-varargs is most likely a source incompatible change:
```java
void test(int... args) {
}
// to:
void test(int[] args) {
}
```
Fails with the following source code:
```java
test(1, 2);
```

Changing a method from non-varargs to varargs is most likely source compatible (unless I am overlooking some corner case with overloads).

Feedback is appreciated!

### CODE REVIEW

File: japicmp/src/main/java/japicmp/compat/CompatibilityChanges.java [hunk 1]
```diff
@@ -265,6 +265,7 @@ private void checkIfConstructorsHaveChangedIncompatible(JApiClass jApiClass, Map
 				}
 			}
 			checkIfAnnotationDeprecatedAdded(constructor);
+			checkIfVarargsChanged(constructor);
```
Comment (Marcono1234) [inline japicmp/src/main/java/japicmp/compat/CompatibilityChanges.java:268]: Have also added this check for constructors, though the names `METHOD_NOW_VARARGS` and `METHOD_NO_LONGER_VARARGS` might be a bit misleading then. Should I rename then?

**Comment by siom79** [comment]
Released with 0.16.0.

Thanks a lot for the contribution.
